src/resultset/resultset_records.c

```diff
@@ -0,0 +1,186 @@
+            if (print_type) RedisModule_ReplyWithStringBuffer(ctx, "integer", 7);
+            return;
+        case T_UINT:
+            RedisModule_ReplyWithLongLong(ctx, v.uintval);
+            if (print_type) RedisModule_ReplyWithStringBuffer(ctx, "integer", 7);
+            return;
+        case T_FLOAT:
+            RedisModule_ReplyWithDouble(ctx, (double)v.floatval);
+            if (print_type) RedisModule_ReplyWithStringBuffer(ctx, "double", 6);
+            return;
+        case T_DOUBLE:
+            RedisModule_ReplyWithDouble(ctx, v.doubleval);
+            if (print_type) RedisModule_ReplyWithStringBuffer(ctx, "double", 6);
+            return;
+        case T_BOOL:
+            if (v.boolval == true) RedisModule_ReplyWithStringBuffer(ctx, "true", 4);
+            else RedisModule_ReplyWithStringBuffer(ctx, "false", 5);
+            if (print_type) RedisModule_ReplyWithStringBuffer(ctx, "boolean", 7);
+            return;
+        case T_NULL:
+            RedisModule_ReplyWithNull(ctx);
+            if (print_type) RedisModule_ReplyWithStringBuffer(ctx, "NULL", 4);
+            return;
+        default:
+            assert("Unhandled value type" && false);
+      }
+}
+
+static void _ResultSet_ReplyWithProperties(RedisModuleCtx *ctx, const GraphEntity *e) {
+    int prop_count = ENTITY_PROP_COUNT(e);
+    RedisModule_ReplyWithArray(ctx, prop_count);
+    // Iterate over all properties stored on entity
+    for (int i = 0; i < prop_count; i ++) {
+        RedisModule_ReplyWithArray(ctx, 3);
+        EntityProperty prop = ENTITY_PROPS(e)[i];
+        // Emit the string key
+        RedisModule_ReplyWithStringBuffer(ctx, prop.name, strlen(prop.name));
+        // Emit the value
+        _ResultSet_ReplyWithScalar(ctx, prop.value, true);
+    }
+}
+
+static void _ResultSet_ReplyWithNode(RedisModuleCtx *ctx, Node *n) {
+    /*  Node reply format:
+     *  [
+     *      ["type", "node"]
+     *      ["id", Node ID (integer)]
+     *      ["label", [label (string or NULL)]]
+     *      ["properties", [[name, value, value type] X N]
+     *  ]
+     */
+    // 4 top-level entities in node reply
+    RedisModule_ReplyWithArray(ctx, 4);
+
+    // ["type", "node"]
+    RedisModule_ReplyWithArray(ctx, 2);
+    RedisModule_ReplyWithStringBuffer(ctx, "type", 4);
+    RedisModule_ReplyWithStringBuffer(ctx, "node", 4);
+
+    // ["id", id (integer)]
+    int id = ENTITY_GET_ID(n);
+    RedisModule_ReplyWithArray(ctx, 2);
+    RedisModule_ReplyWithStringBuffer(ctx, "id", 2);
+    RedisModule_ReplyWithLongLong(ctx, id);
+
+    // ["labels", [label (string)]]
+    RedisModule_ReplyWithArray(ctx, 2);
+    RedisModule_ReplyWithStringBuffer(ctx, "labels", 6);
+    // Print label in nested array for multi-label support
+    RedisModule_ReplyWithArray(ctx, 1);
+    // Retrieve label
+    // TODO Make a more efficient lookup for this string
+    GraphContext *gc = GraphContext_GetFromLTS();
+    const char *label = GraphContext_GetNodeLabel(gc, n);
```

Comment:
Please verify that indeed n->label is NULL, if that's the case i.e. nodes reaching this point will have a NULL label, then this is perfectly fine otherwise use n->label.